Sources/Memcache/MemcacheConnection.swift

```diff
@@ -56,23 +67,63 @@ public actor MemcacheConnection: Service {
 
     private var state: State
 
-    /// Initialize a new MemcacheConnection.
+    /// Initialize a new MemcacheConnection, with an option to specify an ID.
+    /// If no ID is provided, a default value is used.
     ///
     /// - Parameters:
     ///   - host: The host address of the Memcache server.
     ///   - port: The port number of the Memcache server.
     ///   - eventLoopGroup: The event loop group to use for this connection.
-    public init(host: String, port: Int, eventLoopGroup: EventLoopGroup) {
+    ///   - id: The unique identifier for the connection (optional).
+    public init(host: String, port: Int, id: ID? = nil, eventLoopGroup: EventLoopGroup) {
         self.host = host
         self.port = port
+        self.id = id ?? MemcacheConnection.nextID.wrappingIncrementThenLoad(ordering: .sequentiallyConsistent)
         let (stream, continuation) = AsyncStream<StreamElement>.makeStream()
         let bufferAllocator = ByteBufferAllocator()
-        self.state = .initial(
-            eventLoopGroup: eventLoopGroup,
-            bufferAllocator: bufferAllocator,
-            requestStream: stream,
-            requestContinuation: continuation
-        )
+        self.closePromise = eventLoopGroup.next().makePromise(of: Void.self)
+        self.state = .initial(eventLoopGroup: eventLoopGroup, bufferAllocator: bufferAllocator, requestStream: stream, requestContinuation: continuation)
+    }
+
+    deinit {
+        // Fulfill the promise if it has not been fulfilled yet
+        closePromise.fail(MemcacheError(code: .connectionShutdown,
+                                        message: "MemcacheConnection deinitialized without closing",
+                                        cause: nil,
+                                        location: .here()))
+    }
+
+    /// Closes the connection. This method is responsible for properly shutting down
+    /// and cleaning up resources associated with the connection.
+    public nonisolated func close() {
+        Task {
```

Comment:
@fabianfett just wanted to ping you here to show you the problems that we hit when adopting `PooledConnection` on an actor. I am curious to hear what you think about this.

Reply:
Here is also a few notes I had in context to this. 

One reason I have a `Task {}` in both `close` and `onClose` was b/c of a error stating `Actor-isolated instance method 'onClose' cannot be used to satisfy nonisolated protocol requirement`. Xcode by default will give you a fix prompting you to `Add 'nonisolated' to 'onClose' to make this instance method not isolated to the actor` well once you adopt that change you will get a `Actor-isolated property 'some prop' can not be referenced from a non-isolated context` following that error spiral lead us here. from my understanding `onClose` since its part of the actor is `isolated` but PooledConnection expects that its methods to be `nonisolated`, thus we hit this conflict.

Reply:
I think it is important that close is not async, because the whole point of `close` is to trigger the close. We don't want to wait for the channel to be closed 100% here.

We can see this as an issue of the `ConnectionPool` design, or the dreaded we issue of we can't send messages to an actor, without waiting for the response. (which would totally solve the issue here)

Also I want to make explicit, that the `run` pattern here does not play nicely with the ConnectionPool, as the ConnectionFactory expects the Connection only to be returned, once the connection has been established. So you will need to add a `onStarted` callback by necessity. 

I think I come back to my favorite point about structured concurrency:

1. There are multiple sensible ways to structure your concurrency.
2. Those different ways may not play nicely with each other.

Reply:
@FranzBusch I think the issue is not really the actor here, but the question _which_ structure do you follow. All our approaches are structured. :)